### PR TITLE
🎉New Destination: Vertica

### PR DESCRIPTION
**Pull request by dhanjimahto**
## What
Creating a new destination connector for vertica database.
Followed link - https://docs.airbyte.com/connector-development/tutorials/building-a-java-destination

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Please update your PR following https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention

**Comment by marcosmarxm** [comment]
@dhanjimahto please update the PR following https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests/#pull-request-title-convention instructions.

**Comment by dhanjimahto** [comment]
@marcosmarxm i updated PR based on instruction mentioned in 
https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests/#pull-request-title-convention
 Please review it.

**Comment by dhanjimahto** [comment]
Hi Team,

Issue - While running testcases locally some test cases are failing because it is not able to find files "source_config.json","destination_config.json", "destination_catalog.json" during test case execution.

Below are steps to reproduce it on local system & we are attaching exception logs, screenshot & logs, pdf also.

Steps to run integration test cases.
1. Take this PR code on your local system and build it locally (Connector -> destination-vertica).
2. Pull image of vertica from link  & run it through terminal - https://hub.docker.com/r/vertica/vertica-ce
3. Config file attached here.
4.  To run integration test case run below command from airbyte root directory
                  ./gradlew :airbyte-integrations:connectors:destination-vertica:integrationTest
5. You will get test cases failure along with report. we are attaching screenshot of failed test cases along with exception in file named "[Integration test case logs.txt.txt](https://github.com/airbytehq/airbyte/files/11394374/Integration.test.case.logs.txt.txt)"
Test cases failed
[Test results - VerticaDestinationAcceptanceTest.pdf](https://github.com/airbytehq/airbyte/files/11394375/Test.results.-.VerticaDestinationAcceptanceTest.pdf)
[config.txt](https://github.com/airbytehq/airbyte/files/11394397/config.txt)

**Comment by evantahler** [comment]
@marcosmarxm a cursory glance at the code seems OK!  Can you please build this connector and set up a test sandbox for it?  

@dhanjimahto - Can you please add a documentation markdown file for this connector following an example in `docs/destinations` and add an icon please?

**Comment by dhanjimahto** [comment]
@evantahler 
I have added markdown file as well as icon file.
Please review code & let me know anything else need to be done from my end.

**Comment by dhanjimahto** [comment]
Hi Team,
Now we have 100 % test case success rate.
Please see screenshot.

image

**Comment by dhanjimahto** [comment]
> > Hi Team, Now we have 100 % test case success rate. Please see screenshot.
> > image
> 
> Can you push this version to Github?

@marcosmarxm  We got this 100 % passed test case because we manually used hard coded connection to database, which we discussed in our office hour.
Later we discussed in office hour, to have vertica database instance running on cloud & we can connect it using hard coded connection.
We did same thing but now at test case 26 it stuck & not moving further.

We will discuss this in office hour.

**Comment by marcosmarxm** [comment]
> [CLA assistant check](https://cla-assistant.io/airbytehq/airbyte?pullRequest=25682) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=25682) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ dhanjimahto❌ rakeshsimplify3xaccountYou have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=25682) it.

@rakeshsimplify3xaccount @dhanjimahto both contributors must sign the CLA. Please make sure its solved.

**Comment by dhanjimahto** [comment]
Hi Airbyte team,

AS per our discussion, now we are pointing to database running on a cluster & getting 100% test case pass.
Please see below screenshot for reference.
Requesting you to please trigger test case for this branch on CI/CD pipe line.

image

**Comment by dhanjimahto** [comment]
> @dhanjimahto can you add the `metadata.yaml` file check other connectors in the main branch.

Sure, will do & update you

**Comment by dhanjimahto** [comment]
> > @dhanjimahto can you add the `metadata.yaml` file check other connectors in the main branch.
> 
> Sure, will do & update you

I added metadata.yml. Please check

**Comment by dhanjimahto** [comment]
> > > @dhanjimahto can you add the `metadata.yaml` file check other connectors in the main branch.
> > 
> > 
> > Sure, will do & update you
> 
> @marcosmarxm  I added metadata.yml. Please check

**Comment by dhanjimahto** [comment]
> > > > @dhanjimahto can you add the `metadata.yaml` file check other connectors in the main branch.
> > > 
> > > 
> > > Sure, will do & update you
> > 
> > 
> > @marcosmarxm  I added metadata.yml. Please let's us know what next needs to be done.

**Comment by dhanjimahto** [comment]
> > > > > @dhanjimahto can you add the `metadata.yaml` file check other connectors in the main branch.
> > > > 
> > > > 
> > > > Sure, will do & update you
> > > 
> > > 
> > > @marcosmarxm  I added metadata.yml. Please let us know what next needs to be done.

File: airbyte-integrations/connectors/destination-vertica/metadata.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+data:
+  registries:
+    cloud:
+      dockerRepository: airbyte/destination-vertica-strict-encrypt
+      enabled: true
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-vertica/metadata.yml:5]:
```suggestion
      enabled: false
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: docs/integrations/destinations/vertica.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+Each stream will be mapped to a separate table in Vertica. Each table will contain 3 columns:
+
+* `_airbyte_ab_id`: a uuid assigned by Airbyte to each event that is processed. The column type in
+  Vertica is `VARCHAR`.
+* `_airbyte_emitted_at`: a timestamp representing when the event was pulled from the data source.
+  The column type in Vertica is `TIMESTAMP WITH TIME ZONE`.
+* `_airbyte_data`: a json blob representing with the event data. The column type in Vertica
+  is `JSONB`.
+
+
```
Comment (marcosmarxm) [inline docs/integrations/destinations/vertica.md:156]: Add the changelog.

File: airbyte-integrations/connectors/destination-vertica/.dockerignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+*
+!Dockerfile
+!build
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/destination-vertica/.dockerignore:3]: Please move the icon to this folder and rename it to `icon.svg`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
Published https://github.com/airbytehq/airbyte/pull/26718

**Pull request closed without merging**